queries/inko/textobjects.scm

```diff
@@ -0,0 +1,262 @@
+; Classes
+(class) @class.outer
+
+(class
+  body: (class_body
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "class.inner" @_start @_end)))
+
+; Traits
+(trait) @class.outer
+
+(trait
+  body: (trait_body
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "class.inner" @_start @_end)))
+
+; Implementations
+(implement_trait) @class.outer
+
+(implement_trait
+  body: (implement_trait_body
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "class.inner" @_start @_end)))
+
+(reopen_class) @class.outer
+
+(reopen_class
+  body: (reopen_class_body
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "class.inner" @_start @_end)))
+
+; Methods
+(method) @function.outer
+
+(method
+  body: (block
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "function.inner" @_start @_end)))
+
+; Loops
+(while
+  body: (block
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end))) @loop.outer
+
+(loop
+  body: (block
+    .
+    "{"
+    .
+    (_) @_start @_end
+    (_)? @_end
+    .
+    "}"
+    (#make-range! "loop.inner" @_start @_end))) @loop.outer
+
+; Conditionals
```

Comment:
Would you mind adding something like
```
(while
  condition: (_) @conditional.inner)
```
(or an equivalent that you deem most appropriate)?